src/routing/tests/mod.rs

```diff
@@ -390,6 +390,102 @@ async fn handle_consensus_on_online_of_elder_candidate() -> Result<()> {
     Ok(())
 }
 
+#[tokio::test]
+async fn handle_consensus_on_online_of_rejoined_node() -> Result<()> {
+    // Create a section with all elders have age as 32.
+    let sk_set = SecretKeySet::random();
+    let mut nodes: Vec<_> = gen_sorted_nodes(ELDER_SIZE)
+        .into_iter()
+        .map(|node| node.with_age(32))
+        .collect();
+    let elders_info = EldersInfo::new(nodes.iter().map(Node::peer), Prefix::default());
+    let (mut section, section_key_share) = create_section(&sk_set, &elders_info)?;
+
+    // Simulate a peer of age 16 is joined.
+    let existing_peer = create_peer().with_age(16);
+    let member_info = MemberInfo::joined(existing_peer);
+    let member_info = proven(sk_set.secret_key(), member_info)?;
+    let _ = section.update_member(member_info);
```

Comment:
You don't need to simulate the first join and offline. You can create the peer in the `Left` state initially and only test the rejoin itself. This is OK, because we already have tests for joining and leaving. It would also make this test shorter and simpler.

In general, these unit tests should always follow this structure, unless there is a good reason not to:
1. Setup the initial state
2. Execute one command
3. Perform assertions on the resulting commands and/or events